src/pages/RFOX/components/Stake/Bridge/hooks/useRfoxBridge.ts

```diff
@@ -0,0 +1,276 @@
+    return fromBaseUnit(
+      bnOrZero(tradeQuoteQuery.data.unwrap().steps[0].feeData.networkFeeCryptoBaseUnit),
+      sellAsset?.precision ?? 0,
+    )
+  }, [sellAsset?.precision, tradeQuoteQuery.data])
+
+  const networkFeeUserCurrency = useMemo(() => {
+    if (!networkFeeCryptoPrecision) return null
+    return bnOrZero(networkFeeCryptoPrecision).times(feeAssetMarketData.price).toFixed()
+  }, [feeAssetMarketData.price, networkFeeCryptoPrecision])
+
+  const { mutateAsync: handleBridge } = useMutation({
+    mutationFn: async () => {
+      if (
+        !(
+          tradeQuoteQuery.data &&
+          tradeQuoteQuery.data.isOk() &&
+          wallet &&
+          sellAsset &&
+          sellAssetAccountNumber !== undefined
+        )
+      )
+        return
+      const tradeQuote = tradeQuoteQuery.data.unwrap()
+
+      const supportsEIP1559 = supportsETH(wallet) && (await wallet.ethSupportsEIP1559())
+
+      const unsignedTx = await arbitrumBridgeApi.getUnsignedEvmTransaction!({
+        tradeQuote,
+        chainId: sellAsset.chainId,
+        from: fromAccountId(confirmedQuote.sellAssetAccountId).account,
+        stepIndex: 0,
+        supportsEIP1559,
+        slippageTolerancePercentageDecimal: '0',
+      })
+
+      const adapter = assertGetEvmChainAdapter(sellAsset.chainId)
+
+      const buildCustomTxInput = await createBuildCustomTxInput({
+        accountNumber: sellAssetAccountNumber,
+        adapter,
+        data: unsignedTx.data,
+        value: unsignedTx.value,
+        to: unsignedTx.to,
+        wallet,
+      })
+
+      const txId = await buildAndBroadcast({
+        adapter,
+        buildCustomTxInput,
+        receiverAddress: CONTRACT_INTERACTION, // no receiver for this contract call
+      })
+
+      return txId
+    },
+    onSuccess: (txHash: string | undefined) => {
+      if (!txHash) return
+      setL1TxHash(txHash)
+    },
+  })
+
+  const { data: tradeStatus } = useQuery({
+    queryKey: reactQueries.swapper.arbitrumBridgeTradeStatus(l1TxHash, sellAsset?.chainId).queryKey,
+    queryFn:
+      l1TxHash && sellAsset?.chainId
+        ? () =>
+            arbitrumBridgeApi.checkTradeStatus({
+              txHash: l1TxHash,
+              chainId: sellAsset.chainId,
+              quoteId: '',
+              stepIndex: 0,
+            })
+        : skipToken,
+    refetchInterval: 60_000,
+  })
+
+  useEffect(() => {
+    if (!tradeStatus) return
+    if (tradeStatus.message === 'L2 Tx Pending' || tradeStatus.status === TxStatus.Confirmed) {
+      setL2TxHash(tradeStatus.buyTxHash)
+    }
+  }, [tradeStatus])
+
+  return {
+    sellAsset,
+    buyAsset,
+    feeAsset,
+    feeAssetMarketData,
+    sellAssetAccountNumber,
+    tradeQuoteQuery,
+    allowanceContract,
+    bridgeAmountCryptoPrecision,
+    bridgeAmountUserCurrency,
+    networkFeeUserCurrency,
+    handleBridge,
+    serializedL1TxIndex,
+    serializedL2TxIndex,
+  }
```

Comment:
and this guy - I'll just flag all of the occurrences of this for memoization

Reply:
https://github.com/shapeshift/web/pull/7079#discussion_r1634369672